gomock/call.go

```diff
@@ -245,6 +245,8 @@ func (c *Call) SetArg(n int, value interface{}) *Call {
 		// nothing to do
 	case reflect.Slice:
 		// nothing to do
+	case reflect.Map:
```

Comment:
Thanks for the PR, would you mind adding a test that would have been broken before?

Reply:
@codyoss Thanks for your review. I just have updated my PR with unit test. Please help to take a look.